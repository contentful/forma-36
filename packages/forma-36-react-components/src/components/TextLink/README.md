TextLinks communicate actions and linked resources to users.

## When to use

TextLinks are used to display actions that have a lower priority compared to actions that are accessible via a button. They should also be used when linking to URLs, and are preferred over using buttons when used for this purpose.

There are a number of variations of TextLink styles, here is a guide for when to use them:

- **Primary** - Used for primary actions, or the default for links to URLs.
- **Positive** - Used for a positive actions, such as creating or publishing a new entity.
- **Negative** - Used for destructive actions - when something can't be undone. For example, deleting entities.
- **Secondary** - For actions that should be emphasized less than the default primary style.
- **Muted** - For actions that should be emphasized less than the secondary style.
- **White** - For actions appearing on a dark background.

## Best practices

- Use clear, succinct, and accurate labels
- Start TextLink labels with verbs, for example "Add field", not "New field"
- Reduce complexity by using a small number of actions. Too many actions can create confusion when having to decide
