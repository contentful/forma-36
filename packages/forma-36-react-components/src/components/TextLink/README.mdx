---
title: 'TextLink'
type: 'component'
status: 'stable'
slug: /components/text-link/
github: 'https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components/src/components/TextLink'
storybook: 'https://f36-storybook.contentful.com/?path=/story/components-textlink--default'
---

TextLinks communicate actions and linked resources to users.

- [How to use TextLinks](#how-to-use-TextLinks)
- [Component variations](#component-variations)
- [Code examples](#code-examples)
- [Content recommendations](#content-recommendations)
- [Accessibility](#accessibility)

## How to use TextLinks

- TextLinks are used to display actions that have a lower priority compared to actions that are accessible via a button.
- They should also be used when linking to URLs, and are preferred over using buttons when used for this purpose.
- Use clear, succinct, and accurate labels
- Start TextLink labels with verbs, for example "Add field", not "New field"
- Reduce complexity by using a small number of actions. Too many actions can create confusion when having to decide

## Component variations

There are a number of variations of TextLink styles, here is a guide for when to use them:

- **Primary** - Used for primary actions, or the default for links to URLs.
- **Positive** - Used for a positive actions, such as creating or publishing a new entity.
- **Negative** - Used for destructive actions - when something can't be undone. For example, deleting entities.
- **Secondary** - For actions that should be emphasized less than the default primary style.
- **Muted** - For actions that should be emphasized less than the secondary style.
- **White** - For actions appearing on a dark background.

```jsx
<React.Fragment>
  <TextLink linkType="primary">Primary</TextLink>
  <br />
  <TextLink linkType="positive">Positive</TextLink>
  <br />
  <TextLink linkType="negative">Negative</TextLink>
  <br />
  <TextLink linkType="muted">Muted</TextLink>
  <br />
  <TextLink linkType="secondary">Secondary</TextLink>
  <br />
</React.Fragment>
```

## Code examples

Text links communicate actions and highlight linked resources to users.

```jsx
<React.Fragment>
  <TextLink linkType="primary">Text link</TextLink>
</React.Fragment>
```

### TextLink with icon (`icon`)

```jsx
<React.Fragment>
  <TextLink linkType="primary" icon="ExternalLink">
    Primary
  </TextLink>
  <br />
  <TextLink linkType="positive" icon="ExternalLink">
    Positive
  </TextLink>
  <br />
  <TextLink linkType="negative" icon="ExternalLink">
    Negative
  </TextLink>
  <br />
  <TextLink linkType="muted" icon="ExternalLink">
    Muted
  </TextLink>
  <br />
  <TextLink linkType="secondary" icon="ExternalLink">
    Secondary
  </TextLink>
  <br />
</React.Fragment>
```

## Content recommendations

- To make `TextLink` action-oriented, use a verb. For example, "Add field", not "New field"
- If a `TextLink` is used to navigate to a new page, clearly communicate what the user will see when they click on the link. For example, “View teams documentation” vs “Learn about teams”
- Use clear and succinct copy
- If a `TextLink` is at the end of a sentence, don’t add period after the TextLink

## Accessibility

missing
