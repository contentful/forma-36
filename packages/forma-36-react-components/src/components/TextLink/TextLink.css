@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';

.TextLink {
  /* Composes only works on a single selector */
  composes: focus-outline--default from './../../styles/settings/a11y.css';
}

.TextLink,
.TextLink:link {
  display: inline;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: var(--font-stack-primary);
  font-size: var(--font-size-m);
  text-decoration: none;
  background: none;
  transition: color var(--transition-duration-short)
    var(--transition-easing-default);
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  font-weight: var(--font-weight-medium);

  & > span {
    outline: 0;
    display: inherit;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }
}

.TextLink--primary,
.TextLink--primary:link {
  color: var(--color-blue-mid);
}

.TextLink--positive,
.TextLink--positive:link {
  color: var(--color-green-base);
}

.TextLink--negative,
.TextLink--negative:link {
  color: var(--color-red-base);
}

.TextLink--secondary,
.TextLink--secondary:link {
  color: var(--color-text-mid);
}

.TextLink--muted,
.TextLink--muted:link {
  color: var(--color-text-light);
}

.TextLink--white,
.TextLink--white:link {
  color: var(--color-white);

  &:hover,
  &:focus {
    color: var(--color-white);
    opacity: 0.75;
  }
}

.TextLink--disabled,
.TextLink--disabled:link {
  opacity: 0.5;

  &:hover {
    cursor: not-allowed;
  }

  &.TextLink--primary:hover {
    & .TextLink__icon {
      fill: var(--color-blue-base);
    }

    color: var(--color-blue-base);
  }

  &.TextLink--positive:hover {
    & .TextLink__icon {
      fill: var(--color-green-base);
    }

    color: var(--color-green-base);
  }

  &.TextLink--negative:hover {
    & .TextLink__icon {
      fill: var(--color-red-base);
    }

    color: var(--color-red-base);
  }

  &.TextLink--secondary:hover {
    & .TextLink__icon {
      fill: var(--color-text-mid);
    }

    color: var(--color-text-mid);
  }

  &.TextLink--muted:hover {
    & .TextLink__icon {
      fill: var(--color-text-light);
    }

    color: var(--color-text-light);
  }

  &.TextLink--white:hover {
    & .TextLink__icon {
      fill: var(--color-white);
    }

    color: var(--color-white);
  }
}

.TextLink__icon {
  vertical-align: middle;
  transition: fill var(--transition-duration-short)
    var(--transition-easing-default);
  position: relative;
  bottom: 1px; /* For optical alignment of icon with TextLink text */
}

.TextLink:hover .TextLink__icon,
.TextLink:focus .TextLink__icon,
.TextLink:hover .TextLink__icon:link,
.TextLink:focus .TextLink__icon:link {
  fill: var(--color-text-dark);
}

.TextLink__icon-wrapper {
  display: inline;
  margin-right: 4px;
}

.TextLink__icon-wrapper--right {
  display: inline;
  margin-left: 4px;
}
