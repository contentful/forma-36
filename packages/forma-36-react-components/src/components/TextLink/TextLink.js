import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { iconName } from '../Icon/constants';
import Icon from '../Icon';
import TabFocusTrap from '../TabFocusTrap';
import styles from './TextLink.css';

class TextLink extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    href: PropTypes.string,
    disabled: PropTypes.bool,
    testId: PropTypes.string,
    linkType: PropTypes.oneOf([
      'primary',
      'positive',
      'negative',
      'secondary',
      'muted',
    ]),
    onClick: PropTypes.func,
    extraClassNames: PropTypes.string,
    icon: PropTypes.oneOf(Object.keys(iconName)),
  };

  static defaultProps = {
    linkType: 'primary',
    href: undefined,
    testId: 'cf-ui-text-link',
    disabled: false,
    onClick: undefined,
    extraClassNames: undefined,
    icon: undefined,
  };

  renderIcon() {
    const { icon, linkType } = this.props;

    if (!icon) return undefined;

    return (
      <div className={styles['TextLink__icon-wrapper']}>
        <Icon
          icon={icon}
          color={linkType}
          extraClassNames={styles.TextLink__icon}
        />
      </div>
    );
  }

  render() {
    const {
      children,
      href,
      linkType,
      testId,
      onClick,
      disabled,
      extraClassNames,
      icon,
      ...otherProps
    } = this.props;

    const classNames = cn(styles.TextLink, extraClassNames, {
      [styles[`TextLink--${linkType}`]]: linkType,
      [styles['TextLink--disabled']]: disabled,
    });

    const Element = href ? 'a' : 'button';

    return (
      <Element
        className={classNames}
        data-test-id={testId}
        onClick={!disabled ? onClick : null}
        href={!disabled ? href : null}
        disabled={disabled}
        {...otherProps}
      >
        <TabFocusTrap>
          {this.renderIcon()}
          {children}
        </TabFocusTrap>
      </Element>
    );
  }
}

export default TextLink;
