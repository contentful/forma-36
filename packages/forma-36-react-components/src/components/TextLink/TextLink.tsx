import React, { Component, MouseEventHandler } from 'react';
import cn from 'classnames';
import Icon, { IconType } from '../Icon/Icon';
import TabFocusTrap from '../TabFocusTrap';

const styles = require('./TextLink.css');

export type TextLinkType =
  | 'primary'
  | 'positive'
  | 'negative'
  | 'secondary'
  | 'muted';

type IconPositionType = 'right' | 'left';

export type TextLinkProps = {
  children?: React.ReactNode;
  linkType?: TextLinkType;
  href?: string;
  disabled?: boolean;
  testId?: string;
  onClick?: MouseEventHandler<HTMLAnchorElement | HTMLButtonElement>;
  className?: string;
  icon?: IconType;
  text?: string;
  iconPosition?: IconPositionType;
} & typeof defaultProps;

const defaultProps = {
  linkType: 'primary',
  testId: 'cf-ui-text-link',
  disabled: false,
  iconPosition: 'left',
};

export class TextLink extends Component<TextLinkProps> {
  static defaultProps = defaultProps;

  renderIcon(icon: IconType, linkType: TextLinkType) {
    if (!icon) return undefined;

    return (
      <div
        className={
          this.props.iconPosition === 'right'
            ? styles['TextLink__icon-wrapper--right']
            : styles['TextLink__icon-wrapper']
        }
      >
        <Icon icon={icon} color={linkType} className={styles.TextLink__icon} />
      </div>
    );
  }

  render() {
    const {
      children,
      href,
      linkType,
      testId,
      onClick,
      disabled,
      className,
      icon,
      text,
      iconPosition,
      ...otherProps
    } = this.props;

    const classNames = cn(styles.TextLink, className, {
      [styles[`TextLink--${linkType}`]]: linkType,
      [styles['TextLink--disabled']]: disabled,
    });

    const content = (
      <TabFocusTrap>
        {icon && iconPosition === 'left' && this.renderIcon(icon, linkType)}
        {text || children}
        {icon && iconPosition === 'right' && this.renderIcon(icon, linkType)}
      </TabFocusTrap>
    );

    if (href) {
      return (
        <a
          className={classNames}
          data-test-id={testId}
          onClick={
            disabled
              ? e => {
                  e.preventDefault();
                }
              : onClick
          }
          href={href}
          {...otherProps}
        >
          {content}
        </a>
      );
    }

    return (
      <button
        type="button"
        className={classNames}
        data-test-id={testId}
        onClick={!disabled ? onClick : () => {}}
        disabled={disabled}
        {...otherProps}
      >
        {content}
      </button>
    );
  }
}

export default TextLink;
