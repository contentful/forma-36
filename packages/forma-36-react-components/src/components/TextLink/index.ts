export { TextLink } from './TextLink';
export type { TextLinkProps } from './TextLink';
