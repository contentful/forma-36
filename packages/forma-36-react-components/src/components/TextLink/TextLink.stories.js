import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { host } from 'storybook-host';
import { withInfo } from '@storybook/addon-info';

import TextLink from './TextLink';
import { iconName } from '../Icon/constants';

storiesOf('Components|TextLink', module)
  .addDecorator(
    host({
      align: 'center middle',
      cropMarks: false,
    }),
  )
  .add(
    'default',
    withInfo()(() => (
      <TextLink
        href={text('Href', '')}
        onClick={action('onClick')}
        linkType={select(
          'Link Type',
          {
            'Primary (default)': 'primary',
            Positive: 'positive',
            Negative: 'negative',
            Secondary: 'secondary',
            Muted: 'muted',
          },
          'primary',
        )}
        disabled={boolean('Disabled', false)}
        extraClassNames={text('Extra Class Names', '')}
        icon={select('Icon', ['', ...Object.keys(iconName)], undefined)}
      >
        {text('Label (children)', 'Text Link Label')}
      </TextLink>
    )),
  );
