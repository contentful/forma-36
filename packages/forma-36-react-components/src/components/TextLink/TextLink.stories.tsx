import React from 'react';

import TextLink from './TextLink';
import Paragraph from '../Typography/Paragraph';
import notes from './README.md';

export default {
  title: 'Components/TextLink',
  component: TextLink,
  parameters: {
    propTypes: TextLink['__docgenInfo'],
    notes,
  },
};

export const Default = ({ linkLabel, ...args }) => {
  return <TextLink {...args}>{linkLabel}</TextLink>;
};
Default.args = {
  linkLabel: 'This is a text link',
};

export const UsedWithText = () => {
  return (
    <Paragraph>
      Berlin (/bɜːrˈlɪn/; German: [bɛʁˈliːn] (About this soundlisten)) is the{' '}
      <TextLink href="https://www.wikiwand.com/en/Capital_city" target="_blank">
        capital
      </TextLink>{' '}
      and largest city of{' '}
      <TextLink href="https://www.wikiwand.com/en/Germany" target="_blank">
        Germany
      </TextLink>{' '}
      by both area and population. Its 3,769,495 inhabitants as of 31 December
      2019 make it the most populous city of the European Union, according to
      population within city limits.[8] The city is also one of Germany’s 16
      federal states. It is surrounded by the state of{' '}
      <TextLink href="https://www.wikiwand.com/en/Brandenburg" target="_blank">
        Brandenburg
      </TextLink>
      , and contiguous with{' '}
      <TextLink href="https://www.wikiwand.com/en/Potsdam" target="_blanck">
        Potsdam
      </TextLink>
      , Brandenburg’s capital. The two cities are at the center of the
      Berlin-Brandenburg capital region, which is, with about six million
      inhabitants and an area of more than 30,000 km2,[9] Germany’s
      third-largest metropolitan region after the Rhine-Ruhr and Rhine-Main
      regions.
    </Paragraph>
  );
};
