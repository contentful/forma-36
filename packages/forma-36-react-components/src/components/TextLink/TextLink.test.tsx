import React from 'react';
import { shallow, mount } from 'enzyme';
import { axe } from 'jest-axe';
import TextLink from './TextLink';
import { iconName } from '../Icon/constants';

it('renders as a button', () => {
  const output = shallow(<TextLink>Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a link', () => {
  const output = shallow(<TextLink href="#">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a "primary" link type', () => {
  const output = shallow(<TextLink linkType="primary">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a "positive" link type', () => {
  const output = shallow(<TextLink linkType="positive">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a "negative" link type', () => {
  const output = shallow(<TextLink linkType="negative">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a "secondary" link type', () => {
  const output = shallow(<TextLink linkType="secondary">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('renders as a "muted" link type', () => {
  const output = shallow(<TextLink linkType="muted">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('calls an onClick function', () => {
  const onClickFunc = jest.fn();
  const output = shallow(<TextLink onClick={onClickFunc}>Text Link</TextLink>);

  output.simulate('click');

  expect(output).toMatchSnapshot();
  expect(onClickFunc).toHaveBeenCalled();
});

it('prevents onClick function from being called when disabled', () => {
  const onClickFunc = jest.fn();
  const output = shallow(
    <TextLink disabled onClick={onClickFunc}>
      Text Link
    </TextLink>,
  );

  output.simulate('click');

  expect(output).toMatchSnapshot();
  expect(onClickFunc).not.toHaveBeenCalled();
});

it('allows passing additional props not consumed by component', () => {
  const output = shallow(
    <TextLink data-test-id="Testing Id">Text Link</TextLink>,
  );

  expect(output).toMatchSnapshot();
});

it('allows passing additional class names to component', () => {
  const output = shallow(<TextLink className="testing">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('allows passing a test id', () => {
  const output = shallow(<TextLink testId="test-id">Text Link</TextLink>);

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(<TextLink>Text Link</TextLink>).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});

it('has no a11y issues when rendered as a link', async () => {
  const output = mount(<TextLink href="#">Text Link</TextLink>).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});

it('renders with an icon', () => {
  const output = shallow(
    <TextLink icon={iconName[Object.keys(iconName)[0]]}>Text Link</TextLink>,
  );

  expect(output).toMatchSnapshot();
});

it('renders with an icon aligned right to the text', () => {
  const output = shallow(
    <TextLink iconPosition="right" icon={iconName[Object.keys(iconName)[0]]}>
      Text Link
    </TextLink>,
  );

  expect(output).toMatchSnapshot();
});
