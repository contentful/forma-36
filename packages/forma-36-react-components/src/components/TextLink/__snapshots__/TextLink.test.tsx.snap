// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows passing a test id 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="test-id"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`allows passing additional class names to component 1`] = `
<button
  class="TextLink testing TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`allows passing additional props not consumed by component 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="Testing Id"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`calls an onClick function 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`prevents onClick function from being called when disabled 1`] = `
<button
  class="TextLink TextLink--primary TextLink--disabled"
  data-test-id="cf-ui-text-link"
  disabled=""
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "muted" link type 1`] = `
<button
  class="TextLink TextLink--muted"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "negative" link type 1`] = `
<button
  class="TextLink TextLink--negative"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "positive" link type 1`] = `
<button
  class="TextLink TextLink--positive"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "primary" link type 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "secondary" link type 1`] = `
<button
  class="TextLink TextLink--secondary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a "white" link type 1`] = `
<button
  class="TextLink TextLink--white"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a button 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</button>
`;

exports[`renders as a link 1`] = `
<a
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  href="#"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
  </span>
</a>
`;

exports[`renders with an icon 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    <div
      class="TextLink__icon-wrapper"
    >
      <svg
        class="Icon Icon--small Icon--primary TextLink__icon"
        data-test-id="cf-ui-icon"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    Text Link
  </span>
</button>
`;

exports[`renders with an icon aligned right to the text 1`] = `
<button
  class="TextLink TextLink--primary"
  data-test-id="cf-ui-text-link"
  type="button"
>
  <span
    class="TabFocusTrap"
    tabindex="-1"
  >
    Text Link
    <div
      class="TextLink__icon-wrapper--right"
    >
      <svg
        class="Icon Icon--small Icon--primary TextLink__icon"
        data-test-id="cf-ui-icon"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </span>
</button>
`;
